Src/FluentAssertions/Types/PropertyInfoSelector.cs

```diff
@@ -40,16 +40,16 @@ public PropertyInfoSelector(IEnumerable<Type> types)
     }
 
     /// <summary>
-    /// Only select the properties that have a public or internal getter.
+    /// Only select the properties that have a public or internal getter and setter.
     /// </summary>
     public PropertyInfoSelector ThatArePublicOrInternal
     {
         get
         {
             selectedProperties = selectedProperties.Where(property =>
             {
-                MethodInfo getter = property.GetGetMethod(nonPublic: true);
-                return (getter is not null) && (getter.IsPublic || getter.IsAssembly);
+                MethodInfo methodInfo = property.GetGetMethod(nonPublic: true) ?? property.GetSetMethod(nonPublic: true);
+                return (methodInfo is not null) && (methodInfo.IsPublic || methodInfo.IsAssembly);
             });
```

Comment:
The summary now says that _both_ getter and setter must be public/internal, but the implementation only requires one of them.

What about if both a getter and setter exists, but only one is public/internal?
Then there is an asymmetry.

Example
```cs
public class C
{
    public bool PrivateGet { private get; set; } // not included
    public bool PrivateSet { get; private set; } // included
}
```

[SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEUCuA7AHwAEAmARgFgAoIgBgAIiyA6AJRgDMAbGMDASwh4A3NWoA3AIZR6AcxgZ6AXnoYAngAcYEDgAoAwgEpmAcQUAFKBC1R1ugESX+UjDDMZ7hgISiaZAJy65qy68hiGhr5MgQBSQpIAzqEKEb4S0vQJCsqqmtp6RqYWVjZ2jlDOkq4AygqePtTRQSFZ4ZGNAbpxeIm6raliVMAQEFz0wUElMLZqAJJ4HBD0GlMzhtQA3tT0O/RSMgC2CgAWEAAm84s5K9bT6kUY7gCyJ+e6eELmOMBc/GAgqlwMEM9AA/KDlqt7u5ahgXhhTmd3p9vr9/oCcMDfLtGAB2ei6I4I86XJb8BL0D6KPA4LhcEEAMgZBKJiNJzFmCS+Pz+9AIBHorJJCwgHISAEEElkDj81O0qABfQbDUb0bq9Sy3GakyFa9TrKhbKg4oj4m6lNQPZ6vJEfPDctEAjBAkHk+gbeich1/J1AhX0aA7D2cyXS2W+zFK4043b83UWh6w+GI5H21E+jHA+hu4Nc9Po52RgMyd2eiVSmAyrhqCMwBW+KODIgAZkYJHo+k2212LfoKrGThcbmyHpWlVccgUwkyU/oUZNrf74wqQ9hpbC07HQ5nGGnUYVQA=)

```cs
return property.GetGetMethod(nonPublic: true) is { IsPublic: true} or { IsAssembly: true}
    || property.GetSetMethod(nonPublic: true) is { IsPublic: true} or { IsAssembly: true};
```

Reply:
Yes, you're right!
 
 I will change the implemenation based on your suggestion.
 
 Summary: **Only select the properties that have atleast one public or internal accessor**
 
 Any other suggestion for the summary?